MREUnityRuntime/MREUnityRuntimeLib/App/MixedRealityExtensionApp.cs

```diff
@@ -169,6 +173,10 @@ internal MixedRealityExtensionApp(string globalAppId, MonoBehaviour ownerScript,
 				{ typeof(AnimationManager), AnimationManager }
 			});
 
+			_actorManager.RigidBodyAdded += OnRigidBodyAdded;
+			_actorManager.RigidBodyRemoved += OnRigidBodyRemoved;
+			_actorManager.RigidBodyGrabbed += OnRigidBodyGrabbed;
```

Comment:
These need to be deregistered in the shutdown as the app owns the reference to the actor manager, and the actor manager is now getting a reference back to this instance of the app. This will lead to a circular references here that will leak memory.

Reply:
See what we do with the _conn instance in the Disconnect() method.

---
In reply to: [431974794](https://github.com/microsoft/mixed-reality-extension-unity/pull/163#discussion_r431974794) [](ancestors = 431974794)

Reply:
Thanks the tip!
Similar cases handled in ActorManager and Actor.